src/Symfony/Component/Scheduler/Trigger/StaticMessageProvider.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Trigger;
+
+use Symfony\Component\Scheduler\Generator\MessageContext;
+
+final class StaticMessageProvider implements MessageProviderInterface
+{
+    /**
+     * @param array<object> $messages
+     */
+    public function __construct(
+        private array $messages,
+        private string $id = ''
+    ) {
+    }
```

Comment:
```suggestion
    public function __construct(
        public readonly array $messages,
        public readonly string $id = '',
    ) {
    }
```

and then, we can remove the 2 getters?

Reply:
Those methods are implementations of `MessageProviderInterface`

Reply:
Indeed, let's revert the last change and keep private properties. Sorry for the noise.

Reply:
No problem, done